src/System.Management.Automation/engine/LanguagePrimitives.cs

```diff
@@ -3977,7 +3978,9 @@ internal ConvertViaNoArgumentConstructor(ConstructorInfo constructor, Type type)
                     //  - It's in FullLanguage but not because it's part of a parameter binding that is transitioning from ConstrainedLanguage to FullLanguage
                     // When this is invoked from a parameter binding in transition from ConstrainedLanguage environment to FullLanguage command, we disallow
                     // the property conversion because it's dangerous.
-                    if (ecFromTLS == null || (ecFromTLS.LanguageMode == PSLanguageMode.FullLanguage && !ecFromTLS.LanguageModeTransitionInParameterBinding))
+                    if (ecFromTLS == null || 
+                        ((ecFromTLS.LanguageMode == PSLanguageMode.FullLanguage || ecFromTLS.LanguageMode == PSLanguageMode.ConstrainedLanguageAudit) &&
+                         !ecFromTLS.LanguageModeTransitionInParameterBinding))
```

Comment:
This scenario is: the parameter binding happens in constrained language mode for a command that is about to run in the full language mode (`LanguageModeTransitionInParameterBinding` is set to `true` in this scenario).